doc: Added explanation of preceding/following controllers (#2192)
ros-controls/ros2_control: controller_manager/doc/controller_chaining.rst

.. note::

  Controllers that expose the reference interfaces are switched to chained mode only when their reference interfaces are used by other controllers. When their reference interfaces are not used by other controllers, they are switched back to get references from the subscriber.
  However, the controllers that expose the state interfaces are not triggered to chained mode when their state interfaces are used by other controllers.

> .. note::

Reviewer: I just rendered it, '<text>' looks ok (and is used throughout the page already). Should we move the note further up the page, we use preceding controller already above?

Author: I thought about this. My view is that the reader doesn't need to exactly know what *preceding* means in the first few paragraphs. Having a gist is enough. But for the rest of the document, a proper definition is needed.

>   This document uses terms *preceding* and *following* controller. These terms refer to such ordering of controllers that controller A *precedes* controller B iff A's output is connected to B's reference interface. In the example diagram at the beginning of this section, 'diff_drive_controller' *precedes* 'pid left wheel' and 'pid right wheel'. Consequently, 'pid left wheel' and 'pid right wheel' ar controllers *following* after 'diff_drive_controller'.

Reviewer: ```suggestion
  This document uses terms *preceding* and *following* controller. These terms refer to such ordering of controllers that controller A *precedes* controller B if A's output is connected to B's reference interface. In the example diagram at the beginning of this section, 'diff_drive_controller' *precedes* 'pid left wheel' and 'pid right wheel'. Consequently, 'pid left wheel' and 'pid right wheel' are controllers *following* after 'diff_drive_controller'.
```

Author: The academic iff was intentional. It is the mathematical "if and only if". But if you feel that sounds too academic, a normal if could also work ;)

Reviewer 2: I think we can work with the regular `if` in this case. Someone may come around and "fix it" as a typo later on ;)
